File: frontend/src/app/Models/Event.ts
import { Club } from "./Club";

// Event.ts
export interface Event {
    id: number;
    name: string;
    banner: string;
    description: string;
    number_places: number;
    start_date: string;
    end_date: string;
    start_time: string;
    end_time: string;
    status: string;
    club: Club;
}

File: backend/src/main/java/Models/Event.java
package Models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private String Banner;
    private int number_places;
    private LocalDate start_date;
    private LocalDate end_date;
    private LocalTime start_time;
    private LocalTime end_time;
    private String status;

    @ManyToOne
    private Club club;
}